source tree no longer builds without git

## Description

Source tree no longer builds without `git`. https://github.com/rnpgp/rnp/commit/a76dea265406307e0a23f5112ed397e5ad11c566 made git mandatory. Makes sense for developer builds, but distro maintainers need a cleaner environment to avoid loops. Add a method / switch so that the clean tree builds outside of git.

## Steps to Reproduce

1. Get the `0.15.2` tree
2. Build in a root without `git` installed

## Expected Behavior

Build without git installed, or add an option to override.

## Actual Behavior

```
[   33s] + mkdir -p build
[   33s] + cd build
[   33s] + /usr/bin/cmake /home/abuild/rpmbuild/BUILD/rnp-0.15.2/. '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=lib64 -DCMAKE_INSTALL_LIBEXECDIR=/usr/libexec -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -DNDEBUG' '-DCMAKE_CXX_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -DNDEBUG' '-DCMAKE_Fortran_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -DNDEBUG' '-DCMAKE_EXE_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_MODULE_LINKER_FLAGS=-flto=auto -Wl,--as-needed' '-DCMAKE_SHARED_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/lib64/cmake/rnp -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=off
[   33s] -- Found version.txt with 0.15.2
[   33s] CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
[   33s]   Could NOT find Git (missing: GIT_EXECUTABLE)
[   33s] Call Stack (most recent call first):
[   33s]   /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
[   33s]   /usr/share/cmake/Modules/FindGit.cmake:128 (find_package_handle_standard_args)
[   33s]   cmake/version.cmake:94 (find_package)
[   33s]   CMakeLists.txt:43 (determine_version)
```

--- a/cmake/version.cmake
+++ b/cmake/version.cmake
@@ -37,15 +37,6 @@ macro(_git var)
   )
 endmacro()
 
-# call git, store output in var (can not fail)
-macro(git var)
-  _git(${var} ${ARGN})
-  if (NOT _git_ec EQUAL 0)
-    string(REPLACE ";" " " args "${ARGN}")
-    message(FATAL_ERROR "Failed to execute: git ${args}")
-  endif()
-endmacro()
-
 function(extract_version_info version var_prefix)
   # extract the main components
   #   v1.9.0-3-g5b92266+1546836556
@@ -91,7 +82,7 @@ function(determine_version source_dir var_prefix)
     message(STATUS "Found no version.txt.")
   endif()
   # for GIT_EXECUTABLE
-  find_package(Git REQUIRED)
+  find_package(Git)
   # get a description of the version, something like:
   #   v1.9.1-0-g38ffe82        (a tagged release)
   #   v1.9.1-0-g38ffe82-dirty  (a tagged release with local modifications)

